File: Assets/Script/CZL/LuaCallObject.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class LuaCallObject : MonoBehaviour
{
    // Start is called before the first frame update
    void Start()
    {
        XLuasEnv.Instance.LuaDoStrings("CZL/LuaCallObject");
        //XLuasEnv.Instance.LuaDoString1("CZL");

        //Debug.Log(this.GetType().ToString());
    }

    // Update is called once per frame
    void Update()
    {
        XLuasEnv.Instance.FreeEnv();
    }
}

public class NPC
{
    public string name;
    public int HP
    {
        get;
        set;
    }

    public NPC()
    {

    }

    public NPC(string name)
    {
        this.name = name;
    }

    public void PrintMe()
    {
        Debug.Log("Ãû³Æ"+name);
        Debug.Log("HP" + HP);
    }

    public string output()
    {
        return name;
    }
}

File: DataPath/Lua/CZL/LuaCallObject.lua
--lua实例化一个类
--C#
local  obj=CS.NPC()

obj.HP=100;
print(obj.HP)
--print(CS.obj.PrintMe())

local NPC0=CS.NPC("wangzeyuan")
print(NPC0.name)

--表方法调用表成员变量
NPC0:PrintMe()
print(NPC0:output().."第二次打印")





CS.UnityEngine.GameObject("GameObjectLuaCreated")